packages/contracts-bedrock/test/L2Genesis.t.sol

```diff
@@ -0,0 +1,263 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Test } from "forge-std/Test.sol";
+import { L2Genesis } from "scripts/L2Genesis.s.sol";
+import { console2 as console } from "forge-std/console2.sol";
+import { VmSafe } from "forge-std/Vm.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { stdJson } from "forge-std/StdJson.sol";
+import { LibString } from "@solady/utils/LibString.sol";
+import { Constants } from "src/libraries/Constants.sol";
+
+contract L2GenesisTest is Test {
+    L2Genesis genesis;
+
+    function setUp() public {
+        genesis = new L2Genesis();
+        // Note: to customize L1 addresses,
+        // simply pass in the L1 addresses argument for Genesis setup functions that depend on it.
+        // L1 addresses, or L1 artifacts, are not stored globally.
+        genesis.setUp();
+    }
+
+    function tmpfile() internal returns (string memory) {
+        string[] memory commands = new string[](3);
+        commands[0] = "bash";
+        commands[1] = "-c";
+        commands[2] = "mktemp";
+        bytes memory result = vm.ffi(commands);
+        return string(result);
+    }
+
+    function deleteFile(string memory path) internal {
+        string[] memory commands = new string[](3);
+        commands[0] = "bash";
+        commands[1] = "-c";
+        commands[2] = string.concat("rm ", path);
+        vm.ffi(commands);
+    }
+
+    function readJSON(string memory path) internal returns (string memory) {
+        string[] memory commands = new string[](3);
+        commands[0] = "bash";
+        commands[1] = "-c";
+        commands[2] = string.concat("jq < ", path);
+        return string(vm.ffi(commands));
+    }
+
+    function getJSONKeyCount(string memory path) internal returns (uint256) {
+        string[] memory commands = new string[](3);
+        commands[0] = "bash";
+        commands[1] = "-c";
+        commands[2] = string.concat("jq 'keys | length' < ", path, " | xargs cast abi-encode 'f(uint256)'");
+        return abi.decode(vm.ffi(commands), (uint256));
+    }
+
+    // can this become a modifier?
+    function withTempDump(function (string memory) internal f) internal {
+        string memory path = tmpfile();
+        f(path);
+        deleteFile(path);
+    }
+
+    function getAccount(string memory path, string memory key) internal returns (string memory) {
+        string[] memory commands = new string[](3);
+        commands[0] = "bash";
+        commands[1] = "-c";
+        commands[2] = string.concat("jq '.[\"", key, "\"]' < ", path);
+        return string(vm.ffi(commands));
+    }
+
+    // this is slower..
+    function getBalance(string memory account) internal returns (uint256) {
```

Comment:
We should delete the extra unused functions in this file

Reply:
Maybe we should add some addiitonal checks using these functions instead? What ideas did you have for testing in the original draft?

Reply:
The problem was that they were prohibitively slow

Reply:
It will take forever to run if we do something like "iterate over each account and get its balance" so we should say "jq get all accounts that have non zero balance" and "jq get all accounts that have 1 wei" and assert that the number is the same, this will say there are definitely only that number of accounts with 1 wei and it should be the expected number which iirc is 256